pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2034,13 +2034,18 @@ func ipPermissionExists(newPermission, existing *ec2.IpPermission, compareGroupU
 			return false
 		}
 	}
+
 	for _, leftPair := range newPermission.UserIdGroupPairs {
+		found := false
 		for _, rightPair := range existing.UserIdGroupPairs {
 			if isEqualUserGroupPair(leftPair, rightPair, compareGroupUserIDs) {
-				return true
+				found = true
+				break
 			}
 		}
-		return false
+		if found == false {
```

Comment:
Agreed & fixed - thank you!

(BTW this isn't urgent because of how we call this, so I've marked it for 1.8)